### PR TITLE
[Shadow DOM] Modify DOMUtils.get to use query selector rather than getElementById

### PR DESCRIPTION
**Pull request by mdgbayly**
…to support shadow roots

We're trying to get tinymce working in web components that utilize shadow DOM. Our initial usage of tinymce is for an inline editor with a simple toolbar.

My intent is to create a number of small PRs that progressively fix more issues required in order to make our use cases work, rather than one big blanket PR that makes all tinymce use cases support shadow DOM.

This PR enhances DOMUtil.get so that it will use a querySelector rather than getElementById so that it can find elements contained in shadow roots. It uses the root element of the editor as the target for the querySelector.

[Resubmitted to clean up commits and fix assignment to const issue]

### CODE REVIEW

**Comment by TheSpyder** [comment]
You didn't need to open a new PR, force pushing to your branch would have been enough 😄 

I think @Afraithe is the one who knows about our CLA service, I'll ask him about it when I get back to work early next year.

querySelector is quite a bit slower for these operations last time I checked. I'm not sure if it's a performance sensitive area of code, we might need to check that 🤔

File: src/core/main/ts/api/dom/DOMUtils.ts [hunk 1]
```diff
@@ -285,12 +285,13 @@ export function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}
   };
 
   const get = (elm: Node | string): HTMLElement => {
-    if (elm && doc && typeof elm === 'string') {
-      const node = doc.getElementById(elm);
+    const root = settings.root_element || doc;
+    if (elm && root && typeof elm === 'string') {
+      const node = root.querySelector(`#${elm}`) as HTMLElement;
```
Comment (tinydylan) [inline src/core/main/ts/api/dom/DOMUtils.ts:290]: Hi @mdgbayly,

I'm simplifying this function in #5758, which will remove the `getElementsByName` call. The `getElementById` call is still there.

As far as I can tell, `getElementById` is supposed to be valid on a ShadowRoot. Are there browsers that don't support this? I'm having a bit of trouble finding reliable info on this.

Comment (tinydylan) [reply src/core/main/ts/api/dom/DOMUtils.ts:290]: I did a quick test on my Mac, on Firefox 77, Safari 13.1 and Chrome 83 with the following code in the console:

`document.createElement('div').attachShadow({mode: 'open'}).getElementById`

All 3 browsers seem to have this function. Maybe older versions were missing it?

Comment (tinydylan) [reply src/core/main/ts/api/dom/DOMUtils.ts:290]: A better test snippet:

```
var x = document.createElement('div').attachShadow({mode: 'open'}); var y = document.createElement('div'); y.id = 'myid'; x.appendChild(y); x.getElementById('myid');
```

Comment (mdgbayly) [reply src/core/main/ts/api/dom/DOMUtils.ts:290]: Yeh, if that is supported widely it is most likely a better way to do it. That code snippet in my PR may well have come from an earlier PR from another contributor that I pulled a lot of my ideas from, and I may not have thought to check if shadow roots support getElementById.

Looking at the spec it seems to come via ShadowRoot ->DocumentFragment->NonElementParentNode

https://dom.spec.whatwg.org/#interface-nonelementparentnode

Docs for DocumentFragment.getElementById seem to be missing from MDN
https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment

Comment (tinydylan) [reply src/core/main/ts/api/dom/DOMUtils.ts:290]: For now, I think we'll close this PR. If we hit problems with browser support, we'll revisit it.

**Pull request closed without merging**